drivers/flash/nrf_qspi_nor.c

```diff
@@ -1398,38 +1409,34 @@ static int qspi_nor_pm_action(const struct device *dev,
 }
 #endif /* CONFIG_PM_DEVICE */
 
-void  z_impl_nrf_qspi_nor_base_clock_div_force(const struct device *dev,
-					       bool force)
+void z_impl_nrf_qspi_nor_xip_enable(const struct device *dev, bool enable)
 {
-#if defined(CONFIG_SOC_SERIES_NRF53X)
 	struct qspi_nor_data *dev_data = dev->data;
-	/*
-	 * The divider is normally changed, unless the flag is set, only for
-	 * periods when the driver is locked, so the flag itself also can only
-	 * be modified while the driver is locked.
-	 */
+
+	qspi_device_init(dev);
+
+#if NRF_QSPI_HAS_XIPEN
+	nrf_qspi_xip_set(NRF_QSPI, enable);
+#endif
 	qspi_lock(dev);
-	dev_data->keep_base_clock_div_set = force;
+	dev_data->xip_enabled = enable;
 	qspi_unlock(dev);
-#else
-	ARG_UNUSED(dev);
-	ARG_UNUSED(force);
-#endif
+
+	qspi_device_uninit(dev);
```

Comment:
Is it worth setting the `xip_enable` if the device does not seem to support it and the `nrf_qspi_xip_set` has not been invoked?

Reply:
XIP is supported in all nRF SoCs that feature the QSPI peripheral. But only in nRF53 Series it can be disabled in hardware (hence the need to call `nrf_qspi_xip_set()`).